### PR TITLE
fix(table): ensure component is completely rendered when loaded

### PR DESCRIPTION
**Pull request by jensgustafsson**
fix: https://github.com/Lundalogik/crm-feature/issues/1450

Tabulator requires that the html element it's rendered inside
has a size before it's created, otherwise it doesn't consider
it self renderedy completely. (the callback "renderComplete"
is never run).

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/table/table.tsx [hunk 1]
```diff
@@ -175,6 +177,28 @@ export class Table {
...
+     * has a size before it's created, otherwise it doesn't consider
+     * it self renderedy completely. (the callback "renderComplete"
+     * is never run).
+     *
+     * @param table {HTMLElement}
+     * @param options {abulator.Options}
+     *
+     * @returns {void}
+     */
+    private initTabulatorComponent(
```
Comment (Befkadu1) [inline src/components/table/table.tsx:191]: What if we call the name `initTabulatorTable` since `Tabulator` is a class ?

**Comment by Befkadu1** [comment]
The tables not loading on phones. I verified it on browserstack. 
Device: Iphone 11
Safari

Screenshot 2020-10-28 at 09 59 26

**Review by Befkadu1 — CHANGES_REQUESTED** [review]
The tables not loading on phones. This shows only on browserstack. We probably need to check it on the real phone
Device: Iphone 11
Safari
Screenshot 2020-10-28 at 10 41 17

As you see on the log, there is a problem regarding `ResizeObserver`.

**Comment by jensgustafsson** [comment]
Good catch @Befkadu1!

File: src/components/table/table.tsx [hunk 2]
```diff
@@ -175,6 +177,28 @@ export class Table {
...
     }
 
+    /*
+     * Tabulator requires that the html element it's rendered inside
+     * has a size before it's created, otherwise it doesn't consider
+     * it self renderedy completely. (the callback "renderComplete"
+     * is never run).
+     *
+     * @param table {HTMLElement}
+     * @param options {abulator.Options}
```
[the remark this answers is not in the document]
Comment (jensgustafsson) [reply src/components/table/table.tsx]: ⚡

**Comment by jensgustafsson** [comment]
> The tables not loading on phones. This shows only on browserstack. We probably need to check it on the real phone
> Device: Iphone 11
> Safari
> Screenshot 2020-10-28 at 10 41 17
> 
> As you see on the log, there is a problem regarding `ResizeObserver`.

⚡

File: src/components/table/table.tsx [hunk 3]
```diff
@@ -175,6 +177,35 @@ export class Table {
...
+    private initTabulatorComponent(
+        table: HTMLElement,
+        options: Tabulator.Options
+    ) {
+        // Some browsers does not implement the ResizeObserver API...
+        // If that's the case lets just create the table no
+        // matter if its rendered or not.
+        if (!('ResizeObserver' in window)) {
+            this.tabulator = new TabulatorTable(table, options);
+        }
```
Comment (Befkadu1) [inline src/components/table/table.tsx:200]: Should we add `return; ` to avoid the error logs.

Comment (Befkadu1) [reply src/components/table/table.tsx:200]: It means we are going to have a broken table if the browsers don't implement the ResizeObserver API?

File: src/components/table/table.tsx [hunk 4]
```diff
@@ -175,6 +177,35 @@ export class Table {
...
+     * @param table {HTMLElement}
+     * @param options {Tabulator.Options}
+     *
+     * @returns {void}
+     */
+    private initTabulatorComponent(
+        table: HTMLElement,
+        options: Tabulator.Options
+    ) {
+        // Some browsers does not implement the ResizeObserver API...
```
Comment (Befkadu1) [inline src/components/table/table.tsx:195]:
```suggestion
        // Some browsers do not implement the ResizeObserver API...
```

File: src/components/table/table.tsx (hunk 3 above)
Comment (jensgustafsson) [reply src/components/table/table.tsx:200]: You are right!

Yes but it works after you have sorted the table and the problem doesn't seem to appear in the webclient on iphone :)

**Review by Befkadu1 — APPROVED** [review]

**Pull request merged**